### PR TITLE
REFACTOR: random module, backend lib

### PR DESCRIPTION
**Pull request by samir-nasibli**
## Description
* `dpnp_srand` - > `dpnp_rng_srand`
* all cython layer funcs `dpnp_*` to `dpnp_rng_*`
* moved declarations of rng funcs to `dpnp_iface_random.hpp`

### CODE REVIEW

File: dpnp/backend/include/dpnp_iface_random.hpp [hunk 1]
```diff
@@ -0,0 +1,470 @@
...
+
+/**
+ * @defgroup BACKEND_RANDOM_API Backend C++ library interface RANDOM API
+ * @{
+ * This section describes Backend API for RANDOM NUMBER GENERATION (RNG) part.
+ * @}
+ */
+
+/**
+ * @ingroup BACKEND_API
```
Comment (shssf) [inline dpnp/backend/include/dpnp_iface_random.hpp:56]:
```suggestion
 * @ingroup BACKEND_RANDOM_API 

```

File: dpnp/backend/include/dpnp_iface_random.hpp [hunk 2]
```diff
@@ -0,0 +1,470 @@
...
+ */
+template <typename _DataType>
+INP_DLLEXPORT void dpnp_rng_lognormal_c(void* result, _DataType mean, _DataType stddev, size_t size);
+
+/**
+ * @ingroup BACKEND_API
+ * @brief math library implementation of random number generator (multinomial distribution)
+ *
+ * @param [in]  size          Number of elements in `result` arrays.
+ *
```
Comment (shssf) [inline dpnp/backend/include/dpnp_iface_random.hpp:248]: do we need this empty line? Perhaps it is better to remove all these lines from "doctrings" in this file to make it a bit more compact.

Comment (shssf) [reply dpnp/backend/include/dpnp_iface_random.hpp:248]: not sure how it will look in documentation

**Review by shssf — APPROVED** [review]

File: dpnp/backend/include/dpnp_iface_random.hpp (hunk 1 above)
Comment (shssf) [reply dpnp/backend/include/dpnp_iface_random.hpp:56]: to move all these functions into subtopic in documentation

File: dpnp/backend/include/dpnp_iface_random.hpp (hunk 2 above)
Comment (samir-nasibli) [reply dpnp/backend/include/dpnp_iface_random.hpp:248]: @shssf I have checked, you are right. We need refactor this also, see: https://mesos.readthedocs.io/en/latest/doxygen-style-guide/.
We just need to do the same for `dpnp_iface.hpp`. Will do it on other PR.

Comment (shssf) [reply dpnp/backend/include/dpnp_iface_random.hpp]: 👍 nice article.
Please, let's do it step by step. First step inside rng only

**Pull request merged**